src/internal/kinesis.ts

```diff
@@ -0,0 +1,490 @@
+     * @param {JSONObject} options - JSONObject = {}
+     * @returns A RefinedResponse<ResponseType | undefined>
+     */
+    RequestOperation(target: string,
+        options: JSONObject | CreateStreamRequest | PutRecordsRequest | DeleteStreamRequest | Partial<ListStreamRequest> = {}): RefinedResponse<ResponseType | undefined> {
+
+        // handling additional operations as lib standard
+        target = target[0].toUpperCase() + target.slice(1)
+
+        const signedRequest = this.signature.sign(
+            {
+                method: 'POST',
+                protocol: this.awsConfig.scheme,
+                hostname: this.host,
+                path: '/',
+                headers: {
+                    ...this.commonHeaders,
+                    [AMZ_TARGET_HEADER]: `${this.serviceVersion}.${target}`,
+                },
+                body: JSON.stringify(options),
+            },
+            {}
+        )
+        const res = http.request('POST', signedRequest.url, signedRequest.body, {
+            headers: signedRequest.headers,
+        })
+
+        this._handle_error(target, res)
+
+        return res
+    }
+
+    /**
+     * If the response is an error, throw an error
+     * @param {string} operation - The name of the operation that was called.
+     * @param response - RefinedResponse<ResponseType | undefined>
+     * @returns The response is being returned.
+     */
+    _handle_error(
+        operation: string,
+        response: RefinedResponse<ResponseType | undefined>
+    ) {
+        const errorCode = response.error_code
+        if (errorCode === 0) {
+            return
+        }
+
+        const error = response.json() as JSONObject
+        if (errorCode >= 1400 && errorCode <= 1499) {
+            // In the event of certain errors, the message is not set.
+            // Also, note the inconsistency in casing...
+            const errorMessage: string =
+                (error.Message as string) || (error.message as string) || (error.__type as string)
+
+            // Handle specifically the case of an invalid signature
+            if (error.__type === 'InvalidSignatureException') {
+                throw new InvalidSignatureError(errorMessage, error.__type)
+            }
+
+            // Otherwise throw a standard service error
+            throw new KinesisServiceError(errorMessage, error.__type as string, operation)
+        }
+
+        if (errorCode === 1500) {
+            throw new KinesisServiceError(
+                'An error occured on the server side',
+                'InternalServiceError',
+                operation
+            )
+        }
+    }
+
+
+
+    /**
+     * It creates a stream.
+     * @param {CreateStreamRequest} request - CreateStreamRequest - The request object that will be
+     * sent to the server.
+     */
+    createStream(request: CreateStreamRequest): void {
+        this.RequestOperation(`CreateStream`, request)
+    }
+
+    /**
+     * It deletes a stream.
```

Comment:
```suggestion
     * Deletes a Kinesis data stream and all its shards and data.
     *
     * You must shut down any applications operating on the stream before you delete the stream.
     * If an application attempts to operate on a deleted stream, it receives the exception ResourceNotFoundException.
     *
```

Reply:
I'd also like for us to support the `ResourceNotFoundException` explicitly in this function, as described by the AWS documentation. You can find examples of how it's been done in other clients such as `S3Client` or `SecretsManager`. I believe you'll need to define a dedicated error kind, and adjust the `_handle_error` function body to explicitly handle the `ResourceNotFoundException` 👍🏻